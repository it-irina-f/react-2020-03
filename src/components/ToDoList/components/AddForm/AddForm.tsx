import React from "react";
import type { AddFormProps } from "types/todo";

interface Props {
  addListItem: AddFormProps;
}

interface State {
  textInput: string;
}

export class AddForm extends React.Component<Props, State> {
  state = {
    textInput: "",
  };

  submitHandler = (ev: React.FormEvent) => {
    ev.preventDefault();
    this.props.addListItem(this.state.textInput);
  };

  inputChangeHandle = (ev: React.ChangeEvent) => {
    this.setState({
      textInput: (ev.target as HTMLInputElement).value,
    });
  };

  render() {
    return (
      <form onSubmit={this.submitHandler}>
        <input
          type="text"
          name="addlistItem"
          placeholder="Добавить новую задачу"
          value={this.state.textInput}
          onChange={this.inputChangeHandle}
        />

        <button>Добавить</button>
      </form>
    );
  }
}
