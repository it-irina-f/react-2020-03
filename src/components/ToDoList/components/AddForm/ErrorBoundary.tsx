import React from "react";

interface ErrorState {
  hasError: boolean;
}

export class ErrorBoundary extends React.Component<{}, ErrorState> {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  componentDidCatch(error, info) {
    console.log("ошибка", error, info);
  }

  render() {
    if (this.state.hasError) {
      return <h3>Слишком много активных задач</h3>;
    }
    return this.props.children;
  }
}
