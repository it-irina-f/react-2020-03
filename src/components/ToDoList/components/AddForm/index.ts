export * from "./AddForm";
export * from "./ErrorBoundary";
