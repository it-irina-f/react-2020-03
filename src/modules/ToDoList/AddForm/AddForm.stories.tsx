import React from "react";
import { action } from "@storybook/addon-actions";

import { AddForm } from "./AddForm";

export default {
  title: "AddForm Component",
};

export const AddFormComponent = () => [
  <AddForm key="jsx" addListItem={action("add new item in list")} />,
];
