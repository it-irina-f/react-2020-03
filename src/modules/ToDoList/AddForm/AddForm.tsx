import React from "react";
import { Button, Input } from "sancho";
import styled from "@emotion/styled";

type TypeAddForm = (text: string) => void;

interface AddFormProps {
  addListItem: TypeAddForm;
}

interface AddFormState {
  textInput: string;
}

const FormWrapper = styled.form`
  display: flex;
`;

export class AddForm extends React.Component<AddFormProps, AddFormState> {
  state = {
    textInput: "",
  };

  submitHandler = (ev: React.FormEvent) => {
    ev.preventDefault();
    this.props.addListItem(this.state.textInput);
    this.setState({
      textInput: "",
    });
  };

  inputChangeHandle = (ev: React.ChangeEvent) => {
    this.setState({
      textInput: (ev.target as HTMLInputElement).value,
    });
  };

  render() {
    return (
      <FormWrapper onSubmit={this.submitHandler}>
        <Input
          inputSize="md"
          placeholder="Добавить новую задачу"
          type="text"
          name="addlistItem"
          value={this.state.textInput}
          onChange={this.inputChangeHandle}
        />
        <Button intent="primary" type="submit">
          Добавить
        </Button>
      </FormWrapper>
    );
  }
}
