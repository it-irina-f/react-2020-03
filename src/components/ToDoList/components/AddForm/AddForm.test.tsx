import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

import { AddForm } from "./AddForm";

const click = jest.fn();

describe("Test Component AddForm", () => {
  it("renders AddForm", () => {
    expect(
      renderer.create(<AddForm addListItem={click} />).toJSON()
    ).toMatchSnapshot();
  });

/*  it("should be able to click the button", () => {
    const element = mount(<AddForm addListItem={click} />);
    element.find("button").simulate("click");
    expect(click).toHaveBeenCalled();
  });*/
});
