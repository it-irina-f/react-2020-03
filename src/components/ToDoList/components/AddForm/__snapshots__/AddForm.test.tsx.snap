// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component AddForm renders AddForm 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="addlistItem"
    onChange={[Function]}
    placeholder="Добавить новую задачу"
    type="text"
    value=""
  />
  <button>
    Добавить
  </button>
</form>
`;
